CMakeLists.txt

```diff
@@ -101,6 +101,17 @@ if (ENABLE_GOMA)
   set_compiler_launcher(ENABLE_GOMA gomacc)
 endif ()
 
+if(ENABLE_EXTRA_SPEED)
+	if(MSVC)
+		#enable parallel builds
+		add_compile_options("$<$<CXX_COMPILER_ID:MSVC>:/Qpar>")
+		#maximum optimization in Release mode
+		add_compile_options("$<$<AND:$<CXX_COMPILER_ID:MSVC>,$<CONFIG:RELEASE>>:/Ox>")
```

Comment:
Same w/the extra MSVC check here. Please remove.

Also, I would prefer an if instead of a generator exp here. Something like:

if (CMAKE_BUILD_TYPE MATCHES "^Rel")
  add_compile_options(...)
endif ()

That way the flag gets added to Release and RelWithDebInfo while excluding MinSizeRel.

Reply:
Got your point. However, please consider that the progress of compiling a CMake project using Visual Studio is different to CMake + gcc. In case of Visual Studio CMake builts the project once and CMake is not touched anymore afterward (unless you modified CMake files). Switching between Debug, Release etc. is happening in a way that CMake is not involved in this process at all. 

That flag should not be added for MinSizeRel Configurations, because /Ox increases the file size of the binary. In contrast /O1 optimizes for the size of the binary. Current CMake configuration handles this case already correct.
I am also not sure whether it is worth combining /Ox with RelWithDebInfo. The trade of here is higher compilation times vs. consistency with Release build.

Reply:
> Got your point. However, please consider that the progress of compiling a CMake project using Visual Studio is different to CMake + gcc.

Err, yeah, sorry-- I've managed to forget that CMAKE_BUILD_TYPE is kind of useless w/MSVC because of the nature of the MSVC generator. Probably have to use the generator expression here.

> That flag should not be added for MinSizeRel Configurations

MinSizeRel will not match ^Rel; that's why the '^' was there. Moot point though since you probably have to ignore my suggestion and continue with generator use.

Please fix the indentation and comment, and then we can look into getting this merged. 

Also, have you accepted the [Google CLA](https://cla.developers.google.com/clas)? You'll need to do that before I can merge this.